x-pack/platform/plugins/shared/task_manager/server/task.ts

```diff
@@ -483,8 +495,15 @@ export type SerializedConcreteTaskInstance = Omit<
   retryAt: string | null;
   runAt: string;
   partition?: number;
+  userScope?: TaskUserScope;
 };
 
 export type PartialSerializedConcreteTaskInstance = Partial<SerializedConcreteTaskInstance> & {
   id: SerializedConcreteTaskInstance['id'];
 };
+
+export interface ApiKeyOptions {
+  request?: KibanaRequest;
```

Comment:
do we want to allow the ability for users to pass in a pre-generated API key? i believe in an earlier iteration of this, this was possible. do we no longer want to do that?

Reply:
@mikecote and I decided that it might be better if the user just uses the request as they can generate a fake request with their API key. we simply check `user?.authentication_type === 'api_key';` to determine if it's an api key.

Reply:
https://github.com/elastic/kibana/pull/205577#discussion_r1935903213

Reply:
My thinking behind it was that people using our task manager APIs will not know if their `request` object (ex: from an HTTP API call) is authenticated with an API key or something else. It felt easier to let task manager do the determination instead of having every upstream teams determine + extract the API key if the request is authenticated with an API key. I might of missed a scenario in my thinking.